tuf/api/metadata.py

```diff
@@ -335,6 +334,76 @@ def bump_version(self) -> None:
         self.version += 1
 
 
+class Root(Signed):
+    """A container for the signed part of root metadata.
+
+    Attributes:
+        consistent_snapshot: A boolean indicating whether the repository
+            supports consistent snapshots,
+        keys: A dictionary that contains a public key store used to verify
+            top level roles metadata signatures::
+            {
+                '<KEYID>': {
+                    'keytype': '<KEY TYPE>',
+                    'scheme': '<KEY SCHEME>',
+                    'keyid_hash_algorithms': [
+                        '<HASH ALGO 1>',
+                        '<HASH ALGO 2>'
+                        ...
+                    ],
+                    'keyval': {
+                        'public': '<PUBLIC KEY HEX REPRESENTATION>'
+                    }
+                },
+                ...
+            },
+        roles: A dictionary that contains a list of signing keyids and
+            a signature threshold for each top level role::
+            {
+                '<ROLE>': {
+                    'keyids': ['<SIGNING KEY KEYID>', ...],
+                    'threshold': <SIGNATURE THRESHOLD>,
+                },
+                ...
+            }
+
+    """
+    # TODO: determine an appropriate value for max-args and fix places where
+    # we violate that. This __init__ function takes 7 arguments, whereas the
+    # default max-args value for pylint is 5
+    # pylint: disable=too-many-arguments
+    def __init__(
+            self, _type: str, version: int, spec_version: str,
+            expires: datetime, consistent_snapshot: bool,
+            keys: JsonDict, roles: JsonDict) -> None:
+        super().__init__(_type, version, spec_version, expires)
+        # TODO: Add classes for keys and roles
+        self.consistent_snapshot = consistent_snapshot
+        self.keys = keys
+        self.roles = roles
+
+
+    # Serialization.
+    def to_dict(self) -> JsonDict:
+        """Returns the JSON-serializable dictionary representation of self. """
+        json_dict = super().to_dict()
+        json_dict.update({
+            'consistent_snapshot': self.consistent_snapshot,
+            'keys': self.keys,
+            'roles': self.roles
+        })
+        return json_dict
+
+
+    # Modification.
+    def update(self, keys: JsonDict, roles: JsonDict,
+        consistent_snapshot: bool) -> None:
```

Comment:
Now that you've said it, it seems you have a point. I think I like an api of the form:
`root.update('timestamp', key)`

How do you feel about consistent snapshot? Maybe we need a setter/getter pattern for toggling it independently? I am not sure ...

Reply:
Good questions! I think the `update` methods on Targets/Snapshot/Timestamp are supposed to facilitate the  chain of (currently manual) updates that are necessary when the target files in the repository are updated.

Root is not part of that chain, so it may not need to be consistent with the other `update` functions. Regardless, something like `Root.update/delete(role_name, key)` IMHO seems to add more value than mere (multi-)setter logic. 

Generally, I think we should do in these methods whatever is convenient for updating an individual role either manually or from within methods like  `Repository.update_targets(...)`,  `Repository.create_snapshot(...)`, `Repository.rotate_keys(...)` etc., i.e. the methods that actually *do TUF*.

This discussion seems related to https://github.com/theupdateframework/tuf/pull/1193.

Reply:
> Generally, I think we should do in these methods whatever is convenient for updating an individual role either manually or from within methods like `Repository.update_targets(...)`, `Repository.create_snapshot(...)`, `Repository.rotate_keys(...)` etc., i.e. the methods that actually _do TUF_.

I think that sounds like a sensible approach.


I like the `Root.update()` and `Root.remove()` interfaces you've proposed here @sechkova. Do they fulfil the needs which led you to create this PR?

I think the best path forward for this PR would be to implement as much of `Root` functionality as you need to proceed. We can modify/extend the metadata API as we work on making use of it from higher-level pieces of the refactor.

Reply:
I worked on this PR in the context of the client meaning that I only need read access to  'signed' dictionaries.
Anyway I think we all agree on implementing a set of `Root.update()` and `Root.remove()` interfaces which won't cost me much and will be valuable for moving the metadata API forward.